vck-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt

```diff
@@ -251,15 +259,43 @@ class WalletService(
      * decrypting the response if required.
      */
     public suspend fun parseCredentialResponse(
-        response: CredentialResponseParameters,
+        response: String,
+        isEncrypted: Boolean,
+        representation: CredentialRepresentation,
+        scheme: ConstantIndex.CredentialScheme,
+    ): KmmResult<Collection<Holder.StoreCredentialInput>> = catching {
+        response.decryptResponse(isEncrypted)
+            .extractCredentials()
+            .map { encryptionService.decrypt(it).getOrThrow() }
+            .map { it.toStoreCredentialInput(representation, scheme) }
+    }
+
+    private suspend fun String.decryptResponse(encrypted: Boolean) =
+        vckJsonSerializer.decodeFromString<CredentialResponseParameters>(
+            if (encrypted) encryptionService.decrypt(this).getOrThrow() else this
+        )
+
+    /**
+     * Parses [response] received from the credential issuer, mapping to [Holder.StoreCredentialInput],
+     * decrypting the response if required.
+     */
+    public suspend fun parseCredentialResponse(
+        response: CredentialResponse,
         representation: CredentialRepresentation,
         scheme: ConstantIndex.CredentialScheme,
     ): KmmResult<Collection<Holder.StoreCredentialInput>> = catching {
-        response.extractCredentials()
+        response.decryptResponse()
+            .extractCredentials()
             .map { encryptionService.decrypt(it).getOrThrow() }
             .map { it.toStoreCredentialInput(representation, scheme) }
     }
 
+    private suspend fun CredentialResponse.decryptResponse() = when (this) {
```

Comment:
Why's it called `decryptResponse` then the Receiver is a response? I'd suggest `decrypt`, ` decryptIfNeeded`, or something like that